src/main/java/com/education/service/TimetableServiceImpl.java

```diff
@@ -92,48 +91,63 @@ public void update(long id, Timetable newTimetable) {
-     * 3-read a student by getById -->student: Student
-     * 4-add studentId to this List
-     * 6-set student to the timetable
+     * 4-if the size of students in timetable is bigger than capacity of timetable
+     *  4-1- throws an exception
+     * 5-read all students of timetable -->allStudents:List<Students>
+     * 6-read a student by getById -->student: Student
+     * 7-add studentId to this List
+     * 8-set student to the timetable
      * 9-save this timetable
      */
     @Override
     public void addStudentToTimetable(long timetableId, long studentId) {
-        Timetable timetable = getById(timetableId);
+        Timetable target = getById(timetableId);
+        validateStartAndEndOfTimetableWithOtherTimetables(studentId, target);
+        validateTargetTimetableCapacity(target);
+        addStudentToTargetTimetable(studentId, target);
+        timeTableRepository.save(target);
+    }
 
+    public void validateStartAndEndOfTimetableWithOtherTimetables(Long studentId, Timetable target) {
         List<Timetable> timetablesByStudentId = getTimetablesByStudentId(studentId);
 
-        for (Timetable timetables : timetablesByStudentId) {
-            if (timetable.getDate().equals(timetables.getDate())) {
-                if (timetable.getStart().isBefore(timetables.getEnd()) && timetable.getStart().isAfter(timetables.getStart()) || timetable.getEnd()
-                        .isAfter(timetables.getStart()) && timetable.getEnd().isBefore(timetables.getEnd())) {
-                    throw new IllegalArgumentException("This timetable has conflict with other timetables");
-                } else if (timetable.getStart().equals(timetables.getStart())) {
-                    throw new IllegalArgumentException("there is exactly the same timetable");
+        for (Timetable timetable : timetablesByStudentId) {
+            if (target.getDate().equals(timetable.getDate())) {
+
+                boolean startOfTargetBetweenATimetable = target.getStart().isBefore(timetable.getEnd()) && target.getStart().isAfter(timetable.getStart());
+                boolean endOfTargetBetweenAnotherATimetable = target.getEnd().isAfter(timetable.getStart()) && target.getEnd().isBefore(timetable.getEnd());
+
+                if (startOfTargetBetweenATimetable || endOfTargetBetweenAnotherATimetable) {
+                    throw new IllegalArgumentException("This target has conflict with other timetable");
+                } else if (target.getStart().equals(timetable.getStart())) {
+                    throw new IllegalArgumentException("there is exactly the same target");
                 }
             }
         }
-        if (timetable.getStudents().size() > timetable.getCapacity()) {
-            throw new IllegalArgumentException("The capacity of this timetable is full");
+    }
+
+    public void validateTargetTimetableCapacity(Timetable target) {
+        boolean capacityOfStudentsOfTarget = target.getStudents().size() > target.getCapacity();
```

Comment:
1. Try NOT to use "of". Instead, you may rename it to "TargetStudentsCapacity"
2. When I see this variable I expect a number. The variable name is the capacity of students. So it can be 10 or 20. But when I pay attention to variable type it is boolean.
Conclusion: For boolean variables and methods use "is" or "has". For example:
 - hasCapacity
 - isStudentCapacityValid
 - hasCapacityForAnotherStudent
 - and so on...